pgxn/neon/file_cache.c

```diff
@@ -739,6 +740,11 @@ lfc_prewarm(FileCacheState* fcs, uint32 n_workers)
 	n_entries = Min(fcs->n_chunks, lfc_prewarm_limit);
 	Assert(n_entries != 0);
 
+	max_prefetch_pages = n_entries << fcs_chunk_size_log;
+	if (fcs->n_pages > max_prefetch_pages) {
+		elog(ERROR, "LFC: Number of pages in file cache state (%d) is more than the limit (%d)", fcs->n_pages, max_prefetch_pages);
```

Comment:
It is not correct to perform this check outside lock. 
If this check is actually needed, it should be moved (together with calculation of `max_prefetch_pages` under the lock.